drivers/dac/dac_gd32.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2020 BrainCo Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gd_gd32_dac
+
+#include <errno.h>
+#include <drivers/pinctrl.h>
+#include <drivers/dac.h>
+#include <soc.h>
+
+#define LOG_LEVEL CONFIG_DAC_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(dac_gd32);
+
+/**
+ * Most of GD32 series SOCs' DAC have 2 output, DAC0 and DAC1.
+ * But some SOC, like GD32F350 only have one output, DAC0.
+ */
+#if !defined(CONFIG_SOC_GD32F350)
+#define DAC_GD32_OUT_NUM  2
+#define DAC_GD32_OUT_DUAL 1
+#else
+#define DAC_GD32_OUT_NUM  1
+#endif /* !CONFIG_SOC_GD32F350 */
+
+#define DAC_GD32_DEFAULT_VALUE 0
```

Comment:
this can be described in devicetree using a property, e.g. `num-channels` (remember devicetree describes hardware)

Reply:
1. use num-channels dts property to hanndle 2 or 1 DAC output case.
2. use reset-val dts property to hold the default output value.